eZ/Publish/API/Repository/PermissionCriterionResolver.php

```diff
@@ -21,8 +23,10 @@
      *
      * @param string $module
      * @param string $function
+     * @param \eZ\Publish\API\Repository\Values\ValueObject|null $object
+     * @param array $targets
      *
      * @return bool|\eZ\Publish\API\Repository\Values\Content\Query\Criterion
      */
-    public function getPermissionsCriterion($module, $function);
+    public function getPermissionsCriterion($module, $function, ?ValueObject $object = null, array $targets = []);
```

Comment:
hmm, I did not imply ValueObject $object.
There is no value object in this case (yet), as the value is what we match against in the query.

Reply:
I was basically implying:
```
    public function getCriterion(APILimitationValue $value, APIUserReference $currentUser, ?object $target)
    {
        if (in_array($target->id, $value->limitationValues)) {
            return new MatchAll();
        } else {
            return new MatchNone();
        }
    }
```